src/RequestExecutor/HttpRequestExecutor.spec.ts

```diff
@@ -0,0 +1,184 @@
+import 'reflect-metadata';
+
+import { HttpRequestExecutor } from './HttpRequestExecutor';
+import {
+  DefaultVirtualScripts,
+  VirtualScript,
+  VirtualScripts
+} from '../Scripts';
+import { Protocol } from './Protocol';
+import { RequestExecutor } from './RequestExecutor';
+import { Request } from './Request';
+import { RequestExecutorOptions } from './RequestExecutorOptions';
+import { expect, should } from 'chai';
+import nock from 'nock';
+import {
+  anyString,
+  anything,
+  instance,
+  mock,
+  reset,
+  verify,
+  when
+} from 'ts-mockito';
+import { container, Lifecycle } from 'tsyringe';
+
+should();
+
+describe('HttpRequestExecutor', () => {
+  const VirtualScriptsMock = mock<VirtualScripts>(DefaultVirtualScripts);
+  const VirtualScriptMock = mock<VirtualScript>(VirtualScript);
+  const RequestExecutorOptionsMock = mock<RequestExecutorOptions>();
```

Comment:
What is the reason to mock `RequestExecutorOptions`? It makes the tests harder to understand.

Reply:
The reason was that `RequestExecutorOptions` is an interface and I sought it would conceptually right to create a mock. I've rolled it back to a usual object instance.